client/my-sites/plans-grid/components/plan-type-selector/hooks/use-interval-options.tsx

```diff
@@ -0,0 +1,102 @@
+import { UrlFriendlyTermType } from '@automattic/calypso-products';
+import { LocalizeProps, TranslateResult, useTranslate } from 'i18n-calypso';
+import { IntervalTypeProps, SupportedUrlFriendlyTermType } from '../types';
+import generatePath from '../utils';
+import useTermViseMaxDiscounts from './use-term-vise-max-dicounts';
+
+const getDiscountText = ( discountPercentage: number, translate: LocalizeProps[ 'translate' ] ) => {
+	if ( ! discountPercentage ) {
+		return '';
+	}
+	return translate( 'upto %(discount)d% off', {
+		args: { discount: discountPercentage },
+		comment: 'Discount percentage',
+	} );
+};
+export default function useIntervalOptions( props: IntervalTypeProps ): Record<
+	SupportedUrlFriendlyTermType,
+	{
+		key: string;
+		name: TranslateResult;
+		discountText: TranslateResult;
+		url: string;
+		termInMonths: number;
+		ui?: any;
+	}
+> {
+	const translate = useTranslate();
+	const optionList: Record<
+		SupportedUrlFriendlyTermType,
+		{
+			key: string;
+			name: TranslateResult;
+			discountText: TranslateResult;
+			url: string;
+			termInMonths: number;
+		}
+	> = {
+		yearly: {
+			key: 'yearly',
+			name: translate( 'Pay yearly' ),
+			discountText: '',
+			url: '',
+			termInMonths: 12,
+		},
+		'2yearly': {
+			key: '2yearly',
+			name: translate( 'Pay every 2 years' ),
+			discountText: '',
+			url: '',
+			termInMonths: 24,
+		},
+		'3yearly': {
+			key: '3yearly',
+			name: translate( 'Pay every 3 years' ),
+			discountText: '',
+			url: '',
+			termInMonths: 36,
+		},
+		monthly: {
+			key: 'monthly',
+			name: translate( 'Pay monthly' ),
+			discountText: '',
+			url: '',
+			termInMonths: 1,
+		},
+	};
+	const termViseDiscounts = useTermViseMaxDiscounts(
+		props.plans,
+		Object.keys( optionList ) as Array< UrlFriendlyTermType >,
```

Comment:
`NB:`

I think `SupportedUrlFriendlyTermType` is more accurate here?

```suggestion
		Object.keys( optionList ) as Array< SupportedUrlFriendlyTermType >,
```